src/server/plugins/engine/components/ComponentCollection.ts

```diff
@@ -135,4 +196,39 @@ export class ComponentCollection {
 
     return result
   }
+
+  /**
+   * Return error by code for collection fields
+   * For example, 'date.min', 'date.max'
+   */
+  error(helpers: CustomHelpers, code: string, context?: Context) {
+    const { formItems, parent } = this
+    const [field] = formItems
+
+    // Use collection parent name/title when available
+    const key = parent?.name ?? field.name
+    const title = parent?.title ?? field.title
```

Comment:
But similarly to https://github.com/DEFRA/forms-runner/pull/509/files#r1827483136 we also need to locate the parent component so we can pass `{{#title}}` into custom error messages should it be needed:

```js
'{{#title}} must include a {{#label}}'
```